odxtools/minmaxlengthtype.py

```diff
@@ -34,113 +34,136 @@ def __post_init__(self):
-        return termination_char
+                termination_sequence = bytes([0xFF, 0xFF])
+        return termination_sequence
 
     def convert_internal_to_bytes(self, internal_value, encode_state: EncodeState,
                                   bit_position: int) -> bytes:
-        byte_length = self._minimal_byte_length_of(internal_value)
-
-        # The coded value must have at least length min_length
-        if byte_length < self.min_length:
-            raise EncodeError(f"The internal value {internal_value} is only {byte_length} bytes"
-                              f" long but the min length is {self.min_length}")
-        # The coded value must not have a length greater than max_length
-        if self.max_length and byte_length > self.max_length:
-            raise EncodeError(f"The internal value {internal_value} requires {byte_length}"
-                              f" bytes, but the max length is {self.max_length}")
-
-        value_byte = self._to_bytes(
-            internal_value,
-            bit_position=0,
-            bit_length=8 * byte_length,
-            base_data_type=self.base_data_type,
-            is_highlow_byte_order=self.is_highlow_byte_order,
-        )
-
-        if encode_state.is_end_of_pdu or byte_length == self.max_length:
-            # All termination types may be ended by the PDU
-            return value_byte
+        bit_length = 8 * self.max_length if self.max_length is not None else 8 * len(internal_value)
+        bit_length = min(8 * len(internal_value), bit_length)
+        value_bytes = bytearray(
+            self._to_bytes(
+                internal_value,
+                bit_position=0,
+                bit_length=bit_length,
+                base_data_type=self.base_data_type,
+                is_highlow_byte_order=self.is_highlow_byte_order,
+            ))
+
+        # TODO: ensure that the termination delimiter is not
+        # encountered within the encoded value.
+
+        if encode_state.is_end_of_pdu or len(value_bytes) == self.max_length:
+            # All termination types may be ended by the end of the PDU
+            # or once reaching the maximum length. In this case, we
+            # must not add the termination sequence
+            pass
         else:
-            termination_char = self.__termination_character()
-            if self.termination == "END-OF-PDU":
-                termination_char = b''
-            odxassert(
-                termination_char is not None,
-                f"MinMaxLengthType with termination {self.termination}"
-                f"(min: {self.min_length}, max: {self.max_length}) failed encoding {internal_value}"
-            )
-            return value_byte + termination_char
+            termination_sequence = self.__termination_sequence()
+
+            # ensure that we don't try to encode an odd-length
+            # value when using a two-byte terminator
+            odxassert(len(value_bytes) % len(termination_sequence) == 0)
```

Comment:
Are you sure about the `% len(termination_sequence)` thing?
```python
>>> 5 % 0
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ZeroDivisionError: integer modulo by zero
```

Reply:
good point: if there is an error in the ODX file and the termination of such a DOP is set to END-OF-PDU while the corresponding parameter is *not* at the end of the PDU, we get this division by zero. I've added an odxassert: fe6d83870ced635ea36c66337d2974b9ecc59b4a